test/Microsoft.ML.Sweeper.Tests/TestSweeper.cs

```diff
@@ -0,0 +1,756 @@
+            Assert.Equal(normalizedValue, originalNormalizedValue);
+
+            var originalValue = (FloatParameterValue)paramSweep.CreateFromNormalized(normalizedValue);
+            Assert.Equal(originalValue.Value, value.Value);
+        }
+
+        [Theory]
+        [InlineData(0.5, "bar")]
+        [InlineData(0.75, "baz")]
+        public void TestDiscreteValueSweep(double normalizedValue, string expected)
+        {
+            var paramSweep = new DiscreteValueGenerator(new DiscreteParamArguments() { Name = "bla", Values = new[] { "foo", "bar", "baz" } });
+            var value = paramSweep.CreateFromNormalized(normalizedValue);
+            Assert.Equal("bla", value.Name);
+            Assert.Equal(expected, value.ValueText);
+        }
+
+        [Fact]
+        public void TestRandomSweeper()
+        {
+            var args = new SweeperBase.ArgumentsBase();
+            using (var env = new TlcEnvironment(42))
+            {
+                args.SweptParameters = new[]
+                {
+                    ComponentFactoryUtils.CreateFromFunction(
+                        environ => new LongValueGenerator(new LongParamArguments() { Name = "foo", Min = 10, Max = 20 })),
+                    ComponentFactoryUtils.CreateFromFunction(
+                        environ => new LongValueGenerator(new LongParamArguments() { Name = "bar", Min = 100, Max = 200 })),
+                };
+
+                var sweeper = new UniformRandomSweeper(env, args);
+                var initialList = sweeper.ProposeSweeps(5, new List<RunResult>());
+                Assert.Equal(5, initialList.Length);
+                foreach (var parameterSet in initialList)
+                {
+                    foreach (var parameterValue in parameterSet)
+                    {
+                        if (parameterValue.Name == "foo")
+                        {
+                            var val = long.Parse(parameterValue.ValueText);
+                            Assert.True(val >= 10);
+                            Assert.True(val <= 20);
+                        }
+                        else if (parameterValue.Name == "bar")
+                        {
+                            var val = long.Parse(parameterValue.ValueText);
+                            Assert.True(val >= 100);
+                            Assert.True(val <= 200);
+                        }
+                        else
+                        {
+                            Assert.True(false, "Wrong parameter");
+                        }
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void TestSimpleSweeperAsync()
+        {
+            var random = new Random(42);
+            using (var env = new TlcEnvironment(42))
+            {
+                int sweeps = 100;
+
+                var subcomp = new SubComponent<IAsyncSweeper, SignatureAsyncSweeper>("UniformRandomSweeper",
```

Comment:
Can you change this to not use `SubComponent`, but instead just new up the UniformRandomSweeper object directly?